ui/src/components/CsvSourceMapping.vue

```diff
@@ -192,7 +194,13 @@ export default class CsvSourceMapping extends Vue {
   }
 
   async downloadSource (source: CsvSource): Promise<void> {
-    const downloadLink = await this.$store.dispatch('project/downloadCSV', source)
+    const headers = { accept: 'text/csv' }
+    const response = await api.invokeDownloadOperation(source.actions.download, headers)
+    const downloadLink = response.xhr.headers.get('Location')
+
+    if (downloadLink) {
+      window.open(downloadLink)
+    }
```

Comment:
@tpluscode asked me to move the window.open down, but I am fine with this change

Reply:
What do you mean by "move down"?

Reply:
That it lives in the function in the project file (which you deleted completely)

Reply:
👍 

Yeah, I don't agree ;)

Reply:
It is not called `getDownloadLink`, is it? I expected an action which "invokes download" to do exactly that rather than having the UI component look for a HTTP header and call `window.open`

I think it's at least two ways in which this violates boundaries of the involved components

Reply:
I didn't try to change the implementation. But I agree that the flow is a bit weird.